File: sorting/06_selectionSort.js
function selectionSort (A = []) {

  var tempValueIndex;
  var tempValue;

  for (i = 0; i < A.length; i++) {
    tempValueIndex = i;

    for (j = i+1; j < A.length; j++) {
      if(A[j] < A[tempValueIndex]) {
        tempValueIndex = j;
      }
    }

    tempValue = A[i];
    A[i] = A[tempValueIndex];
    A[tempValueIndex] = tempValue;
  }

  return A;
}

File: sorting/06_selectionSort.py
def selectionSort(A):
    valueIndex = 0
    tempValue = 0

    for i in range(0, len(A)):
        valueIndex = i

        for j in range(i+1, len(A)):
            if A[j] < A[valueIndex]:
                valueIndex = j

        tempValue = A[i];
        A[i] = A[valueIndex];
        A[valueIndex] = tempValue;

    return A
